torch/distributed/distributed_c10d.py

```diff
@@ -968,9 +903,32 @@ def _new_process_group_helper(
                 dist_backend_opts.global_ranks_in_group = global_ranks_in_group
 
                 pg = creator_fn(dist_backend_opts, pg_options)
-            _pg_map[pg] = (backend, store)
-            _pg_names[pg] = group_name
 
+        # Process group wrapper initialization for supported PGs when TORCH_DISTRIBUTED_DEBUG is set
+        if backend in [Backend.GLOO, Backend.NCCL, Backend.UCC]:
+            # In debug mode and if GLOO is available, wrap in a wrapper PG that
+            # enables enhanced collective checking for debugability.
+            if get_debug_level() == DebugLevel.DETAIL:
+                if not _GLOO_AVAILABLE:
+                    logger.info(
+                        """TORCH_DISTRIBUTED_DEBUG was set to DETAIL, but
+                                GLOO is not available. Build with Gloo to
+                                create a wrapper process group in debug mode
+                                to aid collective desynchronization debugging."""
+                    )
```

Comment:
Move the current out-most condition as an `elif` here:
```
elif backend not in [Backend.GLOO, Backend.NCCL, Backend.UCC]:
  logger.info(
    """TORCH_DISTRIBUTED_DEBUG was set to DETAIL, but the current backend does not support this feature."""
  )
else:
```

Reply:
Did we want to add a log for this? We currently don't have a message. I am also not sure if there are things other than `ProcessGroupWrapper` that is available when `dist.DebugLevel.DETAIL` is set, maybe the logs are also more verbose so maybe we should just say the current backend does not support `ProcessGroupWrapper`?